Explain proxy and reverse proxy (#5757)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/Administration/Production/ReverseProxy/Index.rst

*   Access control and content filtering
*   Caching frequently requested resources
*   Bypassing geographic restrictions

..  _reverse-proxy-explanation-reverse-proxy:

Difference between proxy and reverse proxy
------------------------------------------

While both types act as intermediaries, they serve different purposes:

*   **Forward Proxy (Standard Proxy)**:

    *   Sits in front of clients
    *   Helps clients access external resources
    *   Primarily client-focused
    *   Masks client identity from servers

*   **Reverse Proxy**:

    *   Sits in front of servers
    *   Handles incoming requests to backend servers
    *   Server-focused
    *   Masks server identity from clients

..  _reverse-proxy-benefits-reverse-proxy:

Benefits of a reverse proxy
---------------------------

Implementing a reverse proxy with TYPO3 offers several advantages:

#.  **Load Balancing**:

    *   Distributes incoming traffic across multiple TYPO3 instances
    *   Ensures high availability and optimal resource utilization

#.  **Enhanced Security**:

    *   Acts as a shield for backend TYPO3 servers
    *   Prevents direct access to backend infrastructure

>     *   Provides additional layer for DDoS protection

Reviewer: Mmh "Provides additional layer" says exactly that?

#.  **SSL/TLS Termination**:

    *   Handles HTTPS encryption/decryption
    *   Reduces computational load on backend TYPO3 servers

> #.  **Caching**:

Author: It is an advantage. It is possible. HOW...is not the question here?

> #.  **Caching**:

Reviewer 2: I've added a section on top of this to highlight "this is optional, if and how you use it is up to you and not the scope of this documentation", hope that's fine.

*   **Protocol Selection**: Determines whether URLs are generated
    with :php:`http://` or :php:`https://`
*   **Redirect Handling**: Controls whether redirects point
    to HTTP or HTTPS URLs

This accurate protocol detection is crucial for maintaining security standards
and ensuring proper functionality across the TYPO3 installation. Incorrect
protocol detection can lead to:

*   Security vulnerabilities due to insecure cookie handling
*   Mixed content warnings in browsers
*   Broken functionality due to incorrect redirect chains
*   Potential security token validation issues

Therefore, proper configuration of reverse proxy settings is essential for
maintaining both security and functionality in TYPO3 installations.

..  _reverse-proxy-https-detection-and-configuration:

HTTPS detection and configuration
---------------------------------

TYPO3 cannot inherently determine whether the original client connection was
HTTP or HTTPS when operating behind a reverse proxy. This is because TYPO3 only
sees the connection between itself and the reverse proxy, which is typically
HTTP and often configured with "HTTPS=on" (e.g., via .htaccess).

> *   If the reverse proxy **exclusively** handles HTTPS connections:

Reviewer: as compared to handling bot http and https as in the sectond bullet point?

> *   If the reverse proxy **exclusively** handles HTTPS connections:

Author: Just one sentence above "which is typically HTTP"

=======================

..  _reverse-proxy-explanation-proxy:

Proxy
-----

A proxy server acts as an intermediary between client devices (such as web
browsers) and the internet. It processes requests from clients seeking resources
from other servers. The primary use cases include:

*   Privacy and anonymity by masking client IP addresses
*   Access control and content filtering
*   Caching frequently requested resources
*   Bypassing geographic restrictions

..  _reverse-proxy-explanation-reverse-proxy:

Difference between proxy and reverse proxy
------------------------------------------

While both types act as intermediaries, they serve different purposes:

*   **Forward Proxy (Standard Proxy)**:

    *   Sits in front of clients
    *   Helps clients access external resources
    *   Primarily client-focused
    *   Masks client identity from servers

*   **Reverse Proxy**:

    *   Sits in front of servers
    *   Handles incoming requests to backend servers
    *   Server-focused
    *   Masks server identity from clients
    *   Influences SSL/TLS termination and security

Choosing the right kind of setup involving proxies is something that falls into the domain

> of your TYPO3 project's DevOps decissions. `https://en.wikipedia.org/wiki/Reverse_proxy`__

Reviewer 2: ```suggestion
of your TYPO3 project's DevOps decissions. `https://en.wikipedia.org/wiki/Reverse_proxy`_
```

> and `https://en.wikipedia.org/wiki/Proxy_server`__ descripe further key factors in depth.

Reviewer 2: ```suggestion
and `https://en.wikipedia.org/wiki/Proxy_server`_ descripe further key factors in depth.
```

While operating a forward proxy requires virtually no TYPO3-specific configuration,
using a Reverse Proxy will require you to configure this integration (additionally to
the specific proxy server configuration, like taking care of SSL/TLS termination):

..  _reverse-proxy-benefits-reverse-proxy:

Benefits of a reverse proxy
---------------------------

> Implementing a reverse proxy with TYPO3 offers several advantages:

Reviewer 2: ```suggestion
Implementing a reverse proxy with TYPO3 can offer several advantages,
depending on its type and configuration. The following list highlights possible
(but not necessary) factors:
```
